mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -114,20 +114,40 @@ class LogisticRegression(override val uid: String)
   def setThreshold(value: Double): this.type = set(threshold, value)
   setDefault(threshold -> 0.5)
 
+  /** @group setParam */
+  def setSampleWeightCol(value: String): this.type = set(sampleWeightCol, value)
+
+  /** @group setParam */
+  def setWeightedSample(value: Boolean): this.type = set(weightedSample, value)
+
   override protected def train(dataset: DataFrame): LogisticRegressionModel = {
     // Extract columns from data.  If dataset is persisted, do not persist oldDataset.
-    val instances = extractLabeledPoints(dataset).map {
-      case LabeledPoint(label: Double, features: Vector) => (label, features)
-    }
+    val instances: Either[RDD[(Double, Vector)], RDD[(Double, Double, Vector)]] =
+      if ($(weightedSample)) {
+        // TODO: Move `setWeightCol` and `extract weight column` code into Predictor class
+        // when we have more algorithms support this feature.
+        Right(dataset.select($(labelCol), $(sampleWeightCol), $(featuresCol)).map {
+          case Row(label: Double, sampleWeight: Double, features: Vector) =>
+            (label, sampleWeight, features)
+        })
+      } else {
+        Left(extractLabeledPoints(dataset).map {
```

Comment:
Okay. This is copying the exiting code for abstract out getting the samples. I will remove it as you suggested.